File: src/Mobile_Customer/lib/features/home/domain/entities/service.dart
class Service {
  final int id;

  final int storeId;
  final String storeName;
  final int categoryId;
  final int? groupId;

  final String name;
  final String? description;
  final String? imageUrl;

  final double price;
  final int durationMinutes;

  final bool isActive;
  final bool isFeatured;
  final int sortOrder;
  final bool isFavorite;

  Service({
    required this.id,
    required this.storeId,
    required this.storeName,
    required this.categoryId,
    this.groupId,
    required this.name,
    this.description,
    this.imageUrl,
    required this.price,
    required this.durationMinutes,
    required this.isActive,
    required this.isFeatured,
    required this.sortOrder,
    required this.isFavorite,
  });
}

File: src/Backend_API/BeautyBookingSystem.Domain/Entities/Service.cs
using BeautyBookingSystem.Domain.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BeautyBookingSystem.Domain.Entities
{
    public class Service : BaseEntity
    {
        public int StoreId { get; set; }
        public virtual Store Store { get; set; } = null!;

        public int CategoryId { get; set; }
        public virtual GlobalCategory Category { get; set; } = null!;

        public int? GroupId { get; set; }
        public virtual ServiceGroup? Group { get; set; }

        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }
        public decimal Price { get; set; }
        public int DurationMinutes { get; set; }
        public bool IsActive { get; set; }
        public bool IsFeatured { get; set; } = false; 
        public int SortOrder { get; set; } = 0;

        public virtual ICollection<BookingDetail> BookingDetails { get; set; } = new List<BookingDetail>();
    }
}
